src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -1,123 +1,240 @@
+	}
+
+	// First try to solve task using java 7 style for processing collections
+
+	/**
+	 * Return road accident with matching index
+	 * 
+	 * @param index
+	 * @return
+	 */
+	public RoadAccident getAccidentByIndex7(String index) {
+		for (RoadAccident roadAccident : roadAccidentList) {
+			if (roadAccident.getAccidentId().equals(index)) {
+				return roadAccident;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * filter list by longtitude and latitude values, including boundaries
+	 * 
+	 * @param minLongitude
+	 * @param maxLongitude
+	 * @param minLatitude
+	 * @param maxLatitude
+	 * @return
+	 */
+	public Collection<RoadAccident> getAccidentsByLocation7(float minLongitude,
+			float maxLongitude, float minLatitude, float maxLatitude) {
+		List<RoadAccident> list = new ArrayList<>();
+		for (RoadAccident roadAccident : roadAccidentList) {
+			if (roadAccident.getLongitude() >= minLongitude
+					&& roadAccident.getLongitude() <= maxLongitude
+					&& roadAccident.getLatitude() >= minLatitude
+					&& roadAccident.getLatitude() <= maxLatitude) {
+				list.add(roadAccident);
+			}
+		}
+		return list;
+	}
+
+	/**
+	 * count incidents by road surface conditions ex: wet -> 2 dry -> 5
+	 * 
+	 * @return
+	 */
+	public Map<String, Long> getCountByRoadSurfaceCondition7() {
+		Map<String, Long> map = new HashMap<>();
+		long dry = 0, wet = 0;
+
+		for (RoadAccident roadAccident : roadAccidentList) {
+			if (roadAccident.getRoadSurfaceConditions().equals("2")
+					|| roadAccident.getRoadSurfaceConditions()
+							.equalsIgnoreCase("wet")) {
+				wet++;
+			} else if (roadAccident.getRoadSurfaceConditions().equals("5")
+					|| roadAccident.getRoadSurfaceConditions()
+							.equalsIgnoreCase("dry")) {
+				dry++;
+			}
+		}
+		map.put("Dry", dry);
+		map.put("Wet", wet);
+		return map;
+	}
+
+	/**
+	 * find the weather conditions which caused the top 3 number of incidents as
+	 * example if there were 10 accidence in rain, 5 in snow, 6 in sunny and 1
+	 * in foggy, then your result list should contain {rain, sunny, snow} - top
+	 * three in decreasing order
+	 * 
+	 * @return
+	 */
+	public List<String> getTopThreeWeatherCondition7() {
+		List<String> list = new LinkedList<>();
+		List<Long> countList = new ArrayList<>();
+		Multimap<String, Long> multiMap = HashMultimap.create();
+		Long count = 0L;
+
+		for (RoadAccident roadAccident : roadAccidentList) {
+			multiMap.put(roadAccident.getWeatherConditions(), count++);
+		}
+
+		for (int i = 0; i < 2; i++) {
+			countList.add(0L);
+		}
+
+		for (String key : multiMap.keySet()) {
+			long size = multiMap.get(key).size();
+			if (size > countList.get(0)) {
+				list.add(0, key);
+				countList.add(0, size);
+			} else if (size > countList.get(1)) {
+				list.add(1, key);
+				countList.add(1, size);
+			} else if (size > countList.get(2)) {
+				list.add(2, key);
+				countList.add(2, size);
+			}
+			if (list.size() == 4) {
+				list.remove(3);
+			}
+		}
+		return list;

```

Comment:
@zero1608 
Can you explain how are you achieving the output from this???
Seems very complex to me.